js/modules/k6/http/request.go

```diff
@@ -340,6 +352,23 @@ func (h *HTTP) parseRequest(
 		}
 	}
 
+	if contentLength := result.Req.Header.Get("Content-Length"); contentLength != "" {
```

Comment:
This particularly has nothing to do with compression it is only about if the user has set `Content-Length` we just set it to the `request.ContentLength`. Technically this can be moved to `MakeRequest` whether it should be in any of those I am not so certain. The code in httpext is only in case we compress the body 

On the other hand I am not refactoring any code I am not adding as this is rabbit hole I am not going in. 

As of now because `ParseRequest` builds the body it has it as a buffer that has len, in `MakeRequest` it is just a `io.ReadCloser` so we can't just move it there.

Reply:
Sounds reasonable, especially the part for not falling into "refactor this stuff right now" rabbit hole :smile: